XHR-lib.pl

```diff
@@ -0,0 +1,48 @@
+#!/usr/local/bin/perl
+# A pile of XHR related routines
+
+use strict;
+
+our (%in, %gconfig, $root_directory, $remote_user, $current_theme);
+
+sub xhr
+{
+
+my %data    = ();
+my $output  = sub {
```

Comment:
Why does this need to be a sub if it's only called once?

Reply:
> Why does this need to be a sub if it's only called once?

This is meant to be called more than once (as code grows), similarly to my `xhr-lib.pl` in the theme.